### PR TITLE
`rustfmt`: Link style guide and embed config options

### PR DESCRIPTION
**Pull request by Urhengulas**

### CODE REVIEW

File: ferrocene/doc/user-manual/src/conf.py [hunk 1]
```diff
@@ -12,13 +12,11 @@
     "ferrocene_domain_cli",
     "ferrocene_autoglossary",
 
+    "myst_parser",
+
     "sphinx.ext.intersphinx",
-    "sphinx.ext.autosectionlabel",
```
Comment (Urhengulas) [inline ferrocene/doc/user-manual/src/conf.py:16]: With `sphinx.ext.autosectionlabel` it fails with:
> /home/urhengulas/Documents/github.com/ferrocene/ferrocene/src/tools/rustfmt/Configurations.md:133: WARNING: duplicate label rustfmt/config:example, other instance in /home/urhengulas/Documents/github.com/ferrocene/ferrocene/ferrocene/doc/user-manual/src/rustfmt/config.rst
> /home/urhengulas/Documents/github.com/ferrocene/ferrocene/src/tools/rustfmt/Configurations.md:253: WARNING: duplicate label rustfmt/config:`"samelinewhere"` (default):, other instance in /home/urhengulas/Documents/github.com/ferrocene/ferrocene/ferrocene/doc/user-manual/src/rustfmt/config.rst
> /home/urhengulas/Documents/github.com/ferrocene/ferrocene/src/tools/rustfmt/Configurations.md:268: WARNING: duplicate label rustfmt/config:`"alwaysnextline"`:, other instance in /home/urhengulas/Documents/github.com/ferrocene/ferrocene/ferrocene/doc/user-manual/src/rustfmt/config.rst

Can we remove it?

**Review by pietroalbini — APPROVED** [review]

File: ferrocene/doc/user-manual/src/conf.py (hunk 1 above)
Comment (pietroalbini) [reply ferrocene/doc/user-manual/src/conf.py:16]: Yeah.

**Comment by Urhengulas** [comment]
bors merge

**Comment by tshepang** [comment]
It is more correct to say *well-formatted* than *well formatted*. Separating the words with space would work better if the words were swapped... *formatted well*.

File: ferrocene/doc/user-manual/src/index.rst
Comment (tshepang) [inline ferrocene/doc/user-manual/src/index.rst:1]: why change `maxdepth` here

File: ferrocene/doc/user-manual/src/rustfmt/usage.rst [hunk 2]
```diff
@@ -6,15 +6,21 @@ Using ``rustfmt``
 
 ``rustfmt`` formats Rust code or checks if Rust code is well formatted.
 
+Code is "well formatted" if it complies with the
+:ref:`formatting rules<configuration>`.
```
Comment (tshepang) [inline ferrocene/doc/user-manual/src/rustfmt/usage.rst:10]:
```suggestion
:ref:`formatting rules <configuration>`.
```

**Comment by Urhengulas** [comment]
> It is more correct to say _well-formatted_ than _well formatted_. Separating the words with space would work better if the words were swapped... _formatted well_.

Changed it to "well-formatted" everywhere.

File: ferrocene/doc/user-manual/src/index.rst
Comment (Urhengulas) [reply ferrocene/doc/user-manual/src/index.rst:1]: I don't think the additional level of headings in the overview makes sense for the "Compilation targets". Don't remember why I did it for "Using rustc", undid that.

**Review by tshepang — APPROVED** [review]
bors merge

File: ferrocene/doc/user-manual/src/rustfmt/usage.rst [hunk 3]
```diff
@@ -30,13 +36,33 @@ If the ``--check`` option is used, ``rusfmt`` checks if the files are well forma
...
+internal error) and ``0`` if formatting completed without error (whether or not
+changes were made).
+
+.. _configuration:
+
+Configuration
+-------------
+
+By default ``rustfmt`` formats code according to the
+`Rust Style Guide <../../../doc/style-guide/index.html>`_. The formatting can
```
Comment (pietroalbini) [inline ferrocene/doc/user-manual/src/rustfmt/usage.rst:52]: Why do you have the `../doc` here?

Comment (Urhengulas) [reply ferrocene/doc/user-manual/src/rustfmt/usage.rst:52]: Good question, probably that is the reason why the link is broken on the published docs (https://ferroussystems.clickup.com/t/8694zvgt3).

Comment (Urhengulas) [reply ferrocene/doc/user-manual/src/rustfmt/usage.rst:52]:
```suggestion
`Rust Style Guide <../../style-guide/index.html>`_. The formatting can
```

**Pull request merged**